packages/xo-server-sdn-controller/src/index.js

```diff
@@ -0,0 +1,443 @@
+import createLogger from '@xen-orchestra/log'
+import { filter, find, map } from 'lodash'
+import forOwn from 'lodash/forOwn'
+import { OVSDBClient } from './ovsdb-client'
+
+const log = createLogger('xo:xo-server:sdn-controller')
+
+const PROTOCOL = 'pssl'
+
+class SDNController {
+  constructor({ xo }) {
+    this._xo = xo
+    this._poolNetworks = []
+    this._OVSDBClients = []
+    this._newHosts = []
+    this._networks = new Map()
+    this._starCenters = new Map()
+  }
+
+  async load() {
+    // FIXME: we should monitor when xapis are added/removed
+    forOwn(this._xo.getAllXapis(), async xapi => {
+      await xapi.objectsFetched
+
+      if (this.setControllerNotNeeded(xapi)) {
+        this.monitorXapi(xapi)
+
+        const hosts = filter(xapi.objects.all, { $type: 'host' })
+        await Promise.all(
+          map(hosts, async host => {
+            this._OVSDBClients.push(new OVSDBClient(host))
+          })
+        )
+
+        // Add already existing pool-wide private networks
+        const networks = filter(xapi.objects.all, { $type: 'network' })
+        forOwn(networks, async network => {
+          if (network.other_config.private_pool_wide === 'true') {
+            log.debug(
+              `Adding network: '${network.name_label}' to managed networks`
+            )
+            const center = await this.electNewCenter(network, true)
+            this._poolNetworks.push({
+              pool: network.$pool.$ref,
+              network: network.$ref,
+              starCenter: center.$ref,
+            })
+            this._networks.set(network.$id, network.$ref)
+            this._starCenters.set(center.$id, center.$ref)
+          }
+        })
+      }
+
+      // TODO: Remove me when UI allows the creation
+      /*
+      this.createPrivateNetwork(
+        xapi.pool,
+        `Private network ${xapi.pool.name_label}`
+      )
+*/
+    })
+  }
+
+  async unload() {
+    while (!this._OVSDBClients.empty) {
+      delete this._OVSDBClients.shift()
+    }
+
+    this._networks.clear()
+  }
```

Comment:
Is the state the same after `load`/`unload` as if the plugin were never loaded?

(e.g. did you remove all listeners?)